lib/gems/pending/util/miq-password.rb

```diff
@@ -72,6 +73,40 @@ def self.encrypted?(str)
     !!split(str).first
   end
 
+  def self.encrypt_hash(values)
+    hash_values = values.deep_clone
+
+    hash_values.select { |f, _| f =~ /[Pp]assword/ }.each do |key, val|
+      next if val.strip.empty?
+
+      key_in_symbol = key.kind_of?(Symbol)
+      hash_values.delete(key)
```

Comment:
can you double check if it is safe to delete/insert an hash entry within an iteration?